src/tacos.mvc/ClientApp/containers/SubmissionContainer.tsx

```diff
@@ -0,0 +1,361 @@
+            });
+        }
+    }
+
+    private checkIsValid = (): boolean => {
+        const { requests } = this.state;
+
+        // make sure we have at least one request
+        if (requests.length === 0) {
+            return false;
+        }
+
+        // make sure all dirty requests are all valid or are being deleted
+        if (!requests.filter(r => r.isDirty).every(r => r.isValid || r.isDeleted || false)) {
+            return false;
+        }
+
+        return true;
+    }
+
+    private submit = async () => {
+        try {
+            const { department } = this.props;
+            const { requests } = this.state;
+
+            // filter just dirty submittions
+            const dirtyRequests = requests.filter(r => r.isDirty);
+
+            // check validity
+            const isValid = this.checkIsValid();
+            if (!isValid) {
+                return;
+            }
+
+            // create the submission
+            const submission: ISubmission = {
+                departmentId: department.id,
+                requests: dirtyRequests
+            };
+
+            const response = await fetch("/requests/submit", {
+                body: JSON.stringify(submission),
+                headers: [["Accept", "application/json"], ["Content-Type", "application/json"]],
+                method: "POST",
+                credentials: "include"
+            });
+
+            if (!response.ok) {
+                throw new Error(response.statusText);
+            }
+
+            const result = await response.json();
+
+            // TODO: make sure we have success
+            LocalStorageService.clearRequests(department);
+            window.location.replace("/requests");
+        } catch (err) {
+            console.error(err);
+        }
+    }
+
+    private focusRequest = (i: number) => {
+        let request = this.state.requests[i];
+        request = { ...request, isFocused: true };
+
+        this.requestUpdated(i, request, false);
+    }
+
+    private removeRequest = (i: number) => {
+        const { department } = this.props;
+        const { requests } = this.state;
+
+        console.log("removing request");
+        let request = this.state.requests[i];
+
+        // if this is an existing request, mark it as deleted, so that we can delete it on the server
+        if (request.id) {
+            request = { ...request, isDeleted: true };
+            this.requestUpdated(i, request);
+            return;
+        }
+
+        // else, remove it from the array
+        const newRequests = requests.filter((r, rIndex) => rIndex !== i);
+        LocalStorageService.saveRequests(department, newRequests);
+        this.setState({ requests: newRequests });
+    }
+
+    private requestUpdated = (i: number, request: IRequest, dirty: boolean = true) => {
+        const { department } = this.props;
+        const { requests } = this.state;
+
+        const annualizationRatio = 4.0 / 12.0;
```

Comment:
@jpknoll would be nice to get this into some simple util since the annualization calculation has changed in the past and I don't want to have to change in too many places or for it to be hard to find.